server/block/tall_grass.go

```diff
@@ -0,0 +1,115 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/item/tool"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+)
+
+// TallGrass is a transparent plant block which can be used to obtain seeds and as decoration.
+type TallGrass struct {
+	replaceable
+	transparent
+	empty
+
+	// Type is the type of grass that the plant represents.
+	Type GrassType
+}
+
+// FlammabilityInfo ...
+func (g TallGrass) FlammabilityInfo() FlammabilityInfo {
+	return newFlammabilityInfo(30, 100, false)
+}
+
+// BreakInfo ...
+func (g TallGrass) BreakInfo() BreakInfo {
+
+	return newBreakInfo(0, alwaysHarvestable, nothingEffective, func(t tool.Tool) []item.Stack {
+		// TODO: Silk touch
+		if t.ToolType() == tool.TypeShears {
+			return []item.Stack{item.NewStack(g, 1)}
+		}
+		if rand.Float32() > 0.57 {
+			return []item.Stack{item.NewStack(WheatSeeds{}, 1)}
+		}
+		return []item.Stack{}
+	})
+}
+
+// BoneMeal attempts to affect the block using a bone meal item.
+func (g TallGrass) BoneMeal(pos cube.Pos, w *world.World) bool {
+	switch g.Type {
+	case NormalGrass():
+		upper := DoubleTallGrass{Type: NormalGrass(), UpperPart: true}
+		if replaceableWith(w, pos.Side(cube.FaceUp), upper) {
+			w.SetBlock(pos, DoubleTallGrass{Type: NormalGrass()})
+			w.SetBlock(pos.Side(cube.FaceUp), upper)
+			return true
+		}
+	case Fern():
+		upper := DoubleTallGrass{Type: Fern(), UpperPart: true}
+		if replaceableWith(w, pos.Side(cube.FaceUp), upper) {
+			w.SetBlock(pos, DoubleTallGrass{Type: Fern()})
+			w.SetBlock(pos.Side(cube.FaceUp), upper)
+			return true
+		}
```

Comment:
These two cases can be merged by using `g.Type`. That's why the `GrassType` is useful!